cmd/release-controller/http_upgrade_graph.go

```diff
@@ -66,40 +76,38 @@ func (c *Controller) graphHandler(w http.ResponseWriter, req *http.Request) {
 					nodes = append(nodes, ReleaseNode{
 						Version: tag.Name,
 						Payload: s.Release.Target.Status.PublicDockerImageRepository + "@" + id,
+						Channel: tag.Annotations[releaseAnnotationName],
 					})
 				} else {
 					nodes = append(nodes, ReleaseNode{
 						Version: tag.Name,
 						Payload: s.Release.Target.Status.PublicDockerImageRepository + ":" + tag.Name,
+						Channel: tag.Annotations[releaseAnnotationName],
 					})
 				}
 			}
 		}
 
 		edges := make([]ReleaseEdge, 0, len(histories))
 		for _, history := range histories {
+			var toChannel = findChannelForNode(history.To, &nodes)
 			switch {
-			case channel == "", channel == "stable":
+			case channel == "", channel == "stable", strings.HasPrefix(channel, "stable-"):
```

Comment:
How many people are actually using these synthetic channel names?  Are OKD clusters dependent on these channels right now (such that changing them breaks people)?

Otherwise, I might suggest we instead have the release controller config contain a field for "additionalChannelNames" where OKD release streams would would have that value (nightly 4.9 is "nightly" and "prerelease") and then you include a version in a channel if the stream describes that name.

That removes all of this hacking around which was really not intended to be expanded this way.

Reply:
>How many people are actually using these synthetic channel names?

```
count by(channel) (max_over_time(cluster_version{version=~".*okd.*"}[7d:15m]) + on(_id) group_left(channel) (0*topk by (channel) (1, cluster_version_available_updates)))
```
says 14 have `stable-4` and 4 `stable`. We generally recommend only `stable-4` channel. Quite a few reports were about "I don't see OKD 4.7 releases when stable-4.6 channel" is selected.

> Are OKD clusters dependent on these channels right now (such that changing them breaks people)?

I don't think we depend on these. CV can be reset by custom payload and we can move everyone to any channel we come up with

> additionalChannelNames

That would work for us, yes